server/src/test/java/org/elasticsearch/action/support/AutoCreateIndexTests.java

```diff
@@ -89,6 +91,12 @@ public void testAutoCreationDisabled() {
         assertEquals("no such index [" + randomIndex + "] and [action.auto_create_index] is [false]", e.getMessage());
     }
 
+    public void testSystemIndexWithAutoCreationDisabled() {
```

Comment:
Another good unit test would be a pattern specifically excluding the system index, e.g. `action.auto_create_index=*,-.foo`. We still want system indices to be auto-created in that case, I think, and it would be good to double check.

Reply:
I've added this.